app/logic/Calendar/RecurrenceRule.ts

```diff
@@ -182,6 +182,17 @@ export class RecurrenceRule implements Readonly<RecurrenceInit> {
     return 'RRULE:' + Object.entries(rule).map(entry => entry.join("=")).join(";");
   }
 
+  /**
+   * Checks whether this rule will invalidate existing occurrences.
+   * We don't check the series length though, as the UI never sets one.
```

Comment:
The logic operates independently from the UI. Please assume that there will be 5 entirely different UIs, all working with this same logic. The UI can and will change at will.

If you need to check the length, please do so.

Reply:
I don't know what you're trying to ask here.

The UI always creates an unending recurrence. But it still wants to know whether any of the other fields changed. So when we're comparing, we can only compare the other fields.